### DISCUSSION TITLE
Are parameters really supported?

### DISCUSSION THREAD

**Discussion by elmart-devo** [memgraph/memgraph#713] [Q&A] [answered]
Hi there. I had a codebase which was sending a lot of similar write queries to memgraph as plain strings. This is, using just `Transaction.run(String query)`. Then, I refactored it to use templates/parameters. This is, using `Transaction.run(String queryTemplate, Map<String, Object> parameters)`. I expected that to boost speed, as in theory, memgraph could reuse query plans for queries sharing the template. But that does not seem to be the case. I observe no meaningful change in speed.

So, my question is: Is memgraph really doing anything different when using the parameters api, or maybe that is not-implemented-yet/not-to-be-implemented?

Thx in advance.

**Comment by antejavor** [comment, marked as answer]
Hi @elmart-devo, sorry for a bit late reply.

So memgraph does support query plan caches. Look here https://memgraph.com/docs/memgraph/reference-guide/configuration#query, meaning if you are executing identical queries all over again, query plans will be reused.

What is important in your case is that it does not matter if you inject parameters via parameters api or use a query as a string because, in the end, it is the same query with the same structure. Memgraph will reuse query structure to cache queries even thou you passed it as a string. Look at the comment above code [piece](https://github.com/memgraph/memgraph/blob/bbfef45b37b3092663aadf0b3e0bb5f6d1f69b88/src/query/cypher_query_interpreter.cpp#L25). In both cases you where measuring performance with query caches, so you probably didn't notice any difference. 

We will document this a bit better. Thanks for asking. cc @vpavicic

**Reply by elmart-devo** [reply]
Ok, got it. Thx for your reply.

**Comment by gitbuda** [comment]
A quick note from my side 😄

Our community has enthusiastically embraced [Discord](https://discord.gg/memgraph) and [Stackoverflow](https://stackoverflow.com/questions/tagged/memgraphdb) as the favored channels for sharing with our team and asking questions. We’ve decided to disable the discussions tab here on GitHub in 2 weeks.

We look forward to seeing you there!
